src/app/tasks/tasks.component.ts

```diff
@@ -1,67 +1,204 @@
-  currentBoard: any;
+  uiPreferences$: any = this.store.select(selectUI);
+  tasks$: Observable<Task[]> = this.store.select(selectAllTasks);
+  board$: Observable<Partial<Board>> = this.store.select(selectBoard);
+  addModal: boolean = false;
+  editModal: boolean = false;
+  addCommentModal: boolean = false;
+
+  showComments = {
+    taskId: '',
+    show: false,
+  };
+
+  currentBoard!: any;
   statusOutput!: string;
   sortOrder: string = '';
   filter: string = '';
   compositeFilter: string = '';
-  // activeTask: string;
-  activeTask!: TaskI;
-  editModal: Observable<boolean> = this.store.select(editTaskSelector);
+  activeTask!: Task;
+
+  columnBcg = {
+    col1: '#c4c2c2',
+    col2: '#c4c2c2',
+    col3: '#c4c2c2',
+  };
+
+  showColorpicker = {
+    col1: false,
+    col2: false,
+    col3: false,
+  };
+
+  icons = {
+    desc: faArrowDownShortWide,
+    asc: faArrowUpWideShort,
+    delete: faTrashCan,
+    edit: faPenToSquare,
+    add: faPlus,
+    paint: faBrush,
+  };
 
   ngOnInit() {
     this.currentBoard = this.activatedRoute.snapshot.paramMap.get('id');
+    this.store.dispatch(getBoardById({ id: this.currentBoard }));
     this.store.dispatch(GetTasks({ id: this.currentBoard }));
+
+    this.uiPreferences$
+      .pipe(
+        skipWhile((res) => !res),
+        tap((res: any) => {
+          this.columnBcg = {
+            col1: res.col1,
+            col2: res.col2,
+            col3: res.col3,
+          };
+        })
+      )
+      .subscribe();
+
+    // this.store.dispatch(GetComments({ boardId: this.currentBoard }));
   }
   constructor(private store: Store, private activatedRoute: ActivatedRoute) {}
 
   openAddTaskModal(value: string) {
     this.statusOutput = value;
-    this.store.dispatch(addTaskOpen());
+    this.addModal = true;
   }
   deleteTask(id: string) {
-    this.store.dispatch(DeleteTask({ id }));
-    console.log(id);
+    this.store.dispatch(deleteTask({ id }));
   }
 
   setSort(order: string, select: string) {
-    return (this.sortOrder = `${order} ${select}`);
+    this.sortOrder = `${order} ${select}`;
   }
   setFilter(param: string) {
     if (this.filter === '' || param === '') {
-      return (this.compositeFilter = '');
+      this.compositeFilter = '';
     }
-    return (this.compositeFilter = `${param} ${this.filter}`);
+    this.compositeFilter = `${param} ${this.filter}`;
   }
 
   onEdit(task: any) {
     this.activeTask = task;
-    this.store.dispatch(editTaskOpen());
+    this.editModal = true;
+  }
+  addComment(todo: any) {
+    this.activeTask = todo;
+    this.addCommentModal = true;
+  }
+
+  toggleShowComments(todo: Task) {
+    this.showComments.taskId = todo.id;
+    this.showComments.show = !this.showComments.show;
+  }
+  conditionShowComments(todo: Task) {
+    if (!todo.userComments.length) {
+      return;
+    }
+    if (todo.userComments.length && todo.id === this.showComments.taskId) {
+      return todo.userComments;
+    }

```

Comment:
if (todo.userComments?.length && todo.id === this.showComments.taskId) {
      return todo.userComments;
    } else {
    return ...
    }